test/bats.bats

```diff
@@ -300,3 +297,44 @@ fixtures bats
   [ "${lines[2]}" = "ok 2 double-quoted name" ]
   [ "${lines[3]}" = "ok 3 unquoted name" ]
 }
+
+@test 'ensure compatibility with unofficial Bash strict mode' {
+  local expected='ok 1 unofficial Bash strict mode conditions met'
+
+  # Run Bats under `set -u` to catch as many unset variable accesses as
+  # possible.
+  run bash -u "${BATS_TEST_DIRNAME%/*}/libexec/bats" \
+    "$FIXTURE_ROOT/unofficial_bash_strict_mode.bats"
+  if [[ "$status" -ne '0' || "${lines[1]}" != "$expected" ]]; then
+    cat <<END_OF_ERR_MSG
+
+This test failed because the Bats internals are violating one of the
+constraints imposed by:
+
+--------
+$(< "$FIXTURE_ROOT/unofficial_bash_strict_mode.bash")
+--------
+
+See:
+- https://github.com/sstephenson/bats/issues/171'
```

Comment:
Trailing `'`

Reply:
Fixed, thanks!

And yes, I'm a huge proponent of fixing even the tiniest details such as this whenever we notice them. The less sloppiness, the less cruft, the less friction, the better experience for everybody!